eng/Versions.props

```diff
@@ -37,4 +38,35 @@
   <PropertyGroup>
     <SystemResourcesExtensionsVersion>4.6.0-preview8.19379.2</SystemResourcesExtensionsVersion>
   </PropertyGroup>
+
+  <Target Name="OverrideArcadeFileVersion"
+          AfterTargets="_InitializeAssemblyVersion">
+    <!-- MSBuild 16.2.0 didn't exist. It shipped as file version 16.200.19.32702,
+         informational version 16.2.32702+c4012a063 because of confusion around
+         Arcade versioning rules: https://github.com/dotnet/arcade/issues/3386
+
+         This caused MSBuild -version to report 16.200.19.32702 and
+         $(MSBuildVersion) to report 16.200.19, a confusing and error-prone
+         situation. Since we're not confident we know all the places the
+         AssemblyFileVersion has an observable effect, we want to set
+         FileVersion to a format like it has been in the past.
+
+         Arcade doesn't support this directly; AutoGenerateAssemblyVersion
+         set to false means that FileVersion=$(AssemblyVersion), but that's
+         not ok for MSBuild because we have a fixed AssemblyVersion for
+         compat (15.1.0.0), but varied FileVersion, which is user-visible
+         via $(MSBuildVersion) and msbuild -version.
+
+         In the future we'll want this to start with $(VersionPrefix), but
+         since 16.2 shipped with a high version, we need to extract a new
+         higher version (generated by Arcade as the fourth part of the
+         version).
+
+         So: we want this to match the NuGet package version and also the
+         AssemblyInformationalVersion. Jump through hoops to do so.
+         -->
+    <PropertyGroup>
+      <FileVersion>$(MSBuildVersionMajorMinor).$(FileVersion.Split('.')[3]).0</FileVersion>
```

Comment:
```suggestion
      <FileVersion>$(VersionPrefix).$(FileVersion.Split('.')[3])</FileVersion>
```

Reply:
I think this is a better format long-term.

Reply:
But the version prefix has all 3 parts. You are ok with that?

Reply:
Yes, I think it makes the most sense to have FileVersion start with `Major.Minor.Revision` and end with the disambiguator.

Reply:
(But I'm open to hearing other thoughts!)